### ISSUE TITLE
[RFC] Fix `enforce_ssl` - two options.

### ISSUE THREAD

**Issue by bobdenotter** [bolt/bolt#6587]
As pointed out by @eduardomart on Slack: Right now, the description for `enforce_ssl` in config.yml says: 

```
# Enforcing the use of SSL. If set, all but the front-end pages will enforce an
# SSL connection (and redirect to HTTPS if you attempt to visit the backend over
# plain HTTP).
enforce_ssl: true
```

This is incorrect. _All_ pages get redirected to their https equivalent, both front- and backend. How should we fix this? 

 - 1) Fix the description in config.yml, so that it matches the behaviour.
 - 2) Change the behaviour, so that it matches the description: Don't redirect to https on pages in the frontend.

**Comment by GawainLynch** [comment]
> Fix the description in config.yml, so that it matches the behaviour.

:+1:

**Comment by bobdenotter** [comment]
Closed by #6589
